### PR TITLE
Public functions

### PR DESCRIPTION
**Pull request by mvondracek**
Implementation of public functions `generate`, `recover`, and `verify`.

Tests in `TestMnemonicPublic` are now set to be skipped. This is until dependencies of public functions are implemented. Tests use test vectors by Trezor from [trezor/python-mnemonic](https://github.com/trezor/python-mnemonic/blob/master/vectors.json).

### CODE REVIEW

**Comment by lsolodkova** [comment]
Where should we check the length of the entropy?

File: PA193_mnemonic_Slytherin/mnemonic.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ def generate(entropy: bytes, seed_password: str = '') -> Tuple[str, bytes]:
     :rtype: Tuple[str, bytes]
     :return: Two item tuple where first is mnemonic phrase and second is seed.
     """
-    pass
+    mnemonic = __entropy2mnemonic(entropy)
```
Comment (lsolodkova) [inline PA193_mnemonic_Slytherin/mnemonic.py]: Need to check the length of the entropy

File: PA193_mnemonic_Slytherin/mnemonic.py [hunk 2]
```diff
@@ -97,17 +99,28 @@ def recover(mnemonic: str, seed_password: str = '') -> Tuple[bytes, bytes]:
...
     :rtype: bool
     :return: True if provided phrase generates expected seed, False otherwise.
     """
-    pass
+    if not __is_valid_mnemonic(mnemonic):
+        raise ValueError('invalid mnemonic')
+    if not __is_valid_seed(expected_seed):
+        raise ValueError('invalid expected_seed')
+
+    entropy, generated_seed = recover(mnemonic, seed_password)
```
Comment (lsolodkova) [inline PA193_mnemonic_Slytherin/mnemonic.py]: Why do we need to compute entropy in the verifying function?

File: PA193_mnemonic_Slytherin/mnemonic.py (hunk 1 above)
Comment (lsolodkova) [inline PA193_mnemonic_Slytherin/mnemonic.py:101]: Need to check entropy length

**Review by lsolodkova — CHANGES_REQUESTED** [review]

File: PA193_mnemonic_Slytherin/mnemonic.py (hunk 1 above)
Comment (mvondracek) [reply PA193_mnemonic_Slytherin/mnemonic.py:101]: Yes, thanks.

File: PA193_mnemonic_Slytherin/mnemonic.py (hunk 2 above)
Comment (mvondracek) [reply PA193_mnemonic_Slytherin/mnemonic.py]: We can only generate seed, sure. :)

**Comment by mvondracek** [comment]
There is also typo in `_secure_seed_compare`, introduced while I worked in multiple branches for PRs.
I will update this PR.

**Review by lsolodkova — APPROVED** [review]

**Review by sobuch — APPROVED** [review]

**Pull request merged**